### PR TITLE
Master stock update quantity on hand dip

### PR DESCRIPTION
**Pull request by dip-odoo**
task:https://www.odoo.com/web#id=30921&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
pad:https://pad.odoo.com/p/r.0eae263fbdb52658839fc928b4aed57f

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sle-odoo** [comment]
Hello, some initial comments

- stock inventory's action_done should now go private since onlyaction_validate is called from the interface
- please make the onchanges of stock inventory and inventory lines private in a specific commit
- adapt test_inventory.py, some tests needs to be adapted now that the tracked products could open a wizard. replace also all call to action_done to action_validate

File: addons/stock/models/stock_inventory.py [hunk 1]
```diff
@@ -54,9 +54,11 @@ def _default_location_id(self):
         readonly=True, required=True,
         states={'draft': [('readonly', False)]},
         default=_default_location_id)
+    product_tmpl_id = fields.Many2one('product.template', 'Product Template')
     product_id = fields.Many2one(
         'product.product', 'Inventoried Product',
         readonly=True,
+        domain=[('type', '=', 'product')],
```
Comment (sle-odoo) [inline addons/stock/models/stock_inventory.py]: a reason to change the domain here? not that it's wrong, but the ui prevents it

File: addons/stock/models/stock_inventory.py [hunk 2]
```diff
@@ -153,6 +158,22 @@ def action_reset_product_qty(self):
         self.mapped('line_ids').write({'product_qty': 0})
         return True
 
+    def action_validate(self):
+        track_line_ids = self.line_ids.filtered(lambda x: x.product_id.tracking in ['lot','serial'] and not x.prod_lot_id)
```
Comment (sle-odoo) [inline addons/stock/models/stock_inventory.py]: - lambda l: l.product_...
- space after the coma
- problematic_tracked_line_ids? not sure of a good variable name.. so you need to add a comment

File: addons/stock/models/stock_inventory.py [hunk 3]
```diff
@@ -153,6 +158,22 @@ def action_reset_product_qty(self):
...
+        if track_line_ids:
+            track_products = track_line_ids.mapped('product_id.product_tmpl_id.name')
+            return {
+                 'name': _('Stock Track Confirmation'),
+                 'type': 'ir.actions.act_window',
+                 'view_type': 'form',
+                 'view_mode': 'form',
+                 'res_model': 'stock.track.confirmaton',
+                 'context': {'default_track_products': ', '.join(track_products), 'default_inventory_id': self.id},
```
Comment (sle-odoo) [inline addons/stock/models/stock_inventory.py]: wouldn't it be better to display in the wizard a one2many with the problematic inventory lines?

File: addons/stock/wizard/stock_track_confirmation.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, tools
+
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_track_confirmation.py]: missing new line

File: addons/stock/models/product.py [hunk 5]
```diff
@@ -517,6 +520,17 @@ def action_view_routes(self):
         action['domain'] = [('id', 'in', routes.ids)]
         return action
 
+    def action_create_inventory_adjustment(self):
```
Comment (sle-odoo) [inline addons/stock/models/product.py]: should this be public? if no, you could make product_id as an argument and simplify the calls

File: addons/stock/wizard/stock_track_confirmation_views.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="view_stock_track_confirmation" model="ir.ui.view">
+        <field name="name">stock.track.confirmaton.view.form</field>
+        <field name="model">stock.track.confirmaton</field>
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_track_confirmation_views.xml]: typo

File: addons/stock/wizard/stock_track_confirmation_views.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        <field name="arch" type="xml">
+            <form string="Confirmation Warning about lot/serial number is empty">
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_track_confirmation_views.xml]: maybe "Lots or serial numbers were not provided to tracked products"

File: addons/stock/models/product.py (hunk 5 above)
Comment (dip-odoo) [reply addons/stock/models/product.py]: yes, method should be public.

File: addons/stock/wizard/stock_track_confirmation_views.xml (hunk 6 above)
Comment (dip-odoo) [reply addons/stock/wizard/stock_track_confirmation_views.xml]: correct it.

File: addons/stock/wizard/stock_track_confirmation_views.xml (hunk 7 above)
Comment (dip-odoo) [reply addons/stock/wizard/stock_track_confirmation_views.xml]: change it

**Comment by jir-odoo** [comment]
LGTM

**Comment by sle-odoo** [comment]
my comments of [last month](https://github.com/odoo/odoo/pull/21464#issuecomment-356009571) weren't taken care of

File: addons/stock/wizard/stock_track_confirmation.py (hunk 4 above)
Comment (dip-odoo) [reply addons/stock/wizard/stock_track_confirmation.py]: it will be done.

Comment (sle-odoo) [reply addons/stock/wizard/stock_track_confirmation.py]: https://github.com/odoo/odoo/pull/21464#issuecomment-356009571

**Pull request merged**